src/app/interaction-model/RequestContext.h

```diff
@@ -0,0 +1,43 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <messaging/ExchangeContext.h>
+
+namespace chip {
+namespace app {
+namespace InteractionModel {
+
+// Context for a currently executing request
+class RequestContext
+{
+public:
+    virtual ~RequestContext() = default;
+
+    /// Valid ONLY during synchronous handling of a Read/Write/Invoke
```

Comment:
But not subscribe?  I'm sure it's valid during sync handling of a subscribe.  

I strongly suggest saying that this is only valid during synchronous handling of an action and leave it at that, without listing examples of actions.